src/liballoc/boxed.rs

```diff
@@ -83,7 +84,7 @@ use crate::str::from_boxed_utf8_unchecked;
 #[lang = "owned_box"]
 #[fundamental]
 #[stable(feature = "rust1", since = "1.0.0")]
-pub struct Box<T: ?Sized>(Unique<T>);
+pub struct Box<T: ?Sized, A: Alloc + Default = Global>(Unique<T>, PhantomData<A>);
```

Comment:
Curiosity: Why include the trait bounds here?  I don't recall anywhere else the libraries do that when the definition of the type doesn't need something from one of the traits.

Also, this new type parameter is insta-stable, right?  Is that a concern?  (Would it work to make the lang item an unstable `CustomBox<T, A>`, and have `type Box<T> = CustomBox<T, Global>;` for now?)

Reply:
> Curiosity: Why include the trait bounds here?

I don't know, that kind of ensures the bounds on all the various impls are right?

> Also, this new type parameter is insta-stable, right? Is that a concern?

Insta-stable, yes. A concern? I don't know.

> (Would it work to make the lang item an unstable `CustomBox<T, A>`, and have `type Box<T> = CustomBox<T, Global>;` for now?)

Unfortunately, the compiler desugars such aliases when printing errors. See https://github.com/rust-lang/rust/pull/50882#issuecomment-402124681 (the part about the extra parameter doesn't apply anymore)